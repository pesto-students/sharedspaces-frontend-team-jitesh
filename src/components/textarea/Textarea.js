import React from 'react'
import './textarea.scss'

const Textarea = ({ name, value, type, id, placeholder, label, onChange, required, className }) => {
    return (
        <div className={`textarea-group mb-4 ${className}`}>
            <label htmlFor={name}>{label}</label>
            <textarea
                onChange={(e) => onChange(e)}
                name={name}
                value={value}
                type={type}
                id={id}
                placeholder={placeholder}
                className='border-2 px-2 py-1.5 rounded transition duration-150 ease-in-out focus:border-gray-500'
                required
            />
        </div>
    )
}

export default Textarea