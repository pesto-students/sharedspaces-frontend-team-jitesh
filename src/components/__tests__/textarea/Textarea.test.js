import { render, screen, cleanup } from "@testing-library/react";
import renderer from 'react-test-renderer'
import Textarea from '../../textarea/Textarea'

afterEach(() => {
    cleanup()
})

test('test matches snapshot', () => {
    const component = renderer.create(<Textarea />)

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot();
})


test('should render textarea component', () => {
    render(<Textarea />)

    const textareaElement = screen.getByTestId("textarea")

    expect(textareaElement).toBeInTheDocument();
})