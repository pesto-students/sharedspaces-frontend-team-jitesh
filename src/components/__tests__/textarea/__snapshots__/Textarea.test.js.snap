// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test matches snapshot 1`] = `
<div
  className="textarea-group mb-4 undefined"
  data-testid="textarea"
>
  <label />
  <textarea
    className="border-2 px-2 py-1.5 rounded transition duration-150 ease-in-out focus:border-gray-500"
    onChange={[Function]}
    required={true}
  />
</div>
`;
